crates/api_crud/src/site/update.rs

```diff
@@ -76,6 +76,12 @@ impl PerformCrud for EditSite {
       }
     }
 
+    if data.private_instance == Some(true) && local_site.federation_enabled {
+      return Err(LemmyError::from_message(
+        "cant_enable_private_instance_if_federation_enabled",
+      ));
+    }
```

Comment:
I think you also need to check the reverse case, with `local_site.private_instance && data.federation_enabled == Some(true)`

Reply:
Good point. I also realized that we should handle the case where the user is changing both (enabling private/disabling federation). Right now, they would have to do one at a time.